velocity/src/main/java/me/leoko/advancedban/velocity/event/RevokePunishmentEvent.java

```diff
@@ -0,0 +1,32 @@
+package me.leoko.advancedban.velocity.event;
+
+import me.leoko.advancedban.utils.Punishment;
+
+public class RevokePunishmentEvent {
+  private final Punishment punishment;
+  private final boolean massClear;
+
+  public RevokePunishmentEvent(Punishment punishment, boolean massClear) {
+    this.punishment = punishment;
```

Comment:
It would be a good idea to validate that `Punishment` is non-null, e.g. using Objects.requireNonNull.

Same goes for PunishmentEvent